### DISCUSSION TITLE
`include` flag imitation?

### DISCUSSION THREAD

**Discussion by woojiq** [sharkdp/fd#1297] [Q&A]
**TL;DR;** How to simulate this: `fd --include=".config"`?
I want to show all files from `fd` but also include files from `.config` folder (if present in current path), is this possible? Of course, I can use `fd -H` but there are a lot of useless dotfiles for me, I need only `.config` dir. There is an "exclude" flag, but no "include" flag. I tried to implement this with a regular expression, but my lack of knowledge does not allow me to do so. :crying_cat_face:

**Comment by wellkilo** [comment]
Use --hidden to allow traversal into .config, then exclude the other hidden entries. fd has no inverse --include override for hidden filtering, because a hidden directory must be traversed before a later match can be found.

For exactly the .config tree under the current directory:

    fd --hidden . .config

or search a particular pattern there:

    fd --hidden PATTERN .config

If you want normal visible results everywhere plus .config, combine two roots/commands:

    { fd PATTERN .; fd --hidden PATTERN .config; } | sort -u

For interactive tools this is often sufficient without sorting. Another single-command option is --hidden plus exclusions for hidden directories you never want, but maintaining a complete negative list is less robust:

    fd --hidden --exclude .git --exclude .cache PATTERN

The key point is that the search root is independent of the match pattern. Passing .config as a root prevents fd from walking every other hidden directory. The README documents multiple search paths in the usage form fd [pattern [path]...] and the hidden/exclude flags: [usage](https://github.com/sharkdp/fd/blob/9e8927e8f3ee7bb699df55cb0b440f848dce6857/README.md#how-to-use).
